common/src/main/java/com/wynntils/features/chat/GuildRankReplacementFeature.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.chat;
+
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.CHAT)
+public class GuildRankReplacementFeature extends Feature {
+    @RegisterConfig
+    public final Config<RankType> rankType = new Config<>(RankType.STARS);
+
+    // Test suite: https://regexr.com/7e5gr
+    private static final Pattern RANK_STARS_PATTERN = Pattern.compile("(§3\\[(?:§b)?)(★{0,5})((?:§3)?.{1,16}])");
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChatMessageReceived(ChatMessageReceivedEvent e) {
+        if (rankType.get() == RankType.STARS) return; // default wynn behavior is stars
+
+        Matcher m = RANK_STARS_PATTERN.matcher(
+                StyledText.fromComponent(e.getMessage()).getString());
```

Comment:
This is not how you are meant to use `StyledText` (`getString` would be private in a better world, Check out `ChatMentionsFeature`). TLDR use `StyledText#getMatcher`.

Reply:
This comment is overall true for the other parts of the replacement, I think you can get away by changing only the correct `StyledTextPart` instead of making a whole new component.